errdefs/unsupported.go

```diff
@@ -0,0 +1,65 @@
+package errdefs
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrUnsupported is an error interface which denotes whether the operation failed due
+// to the operation being unsupported. It will not succeed on subsequent retries.
+type ErrUnsupported interface {
```

Comment:
Pedantic request... WDYT about renaming this to `NotImplemented`?

Otherwise this LGTM.